clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/ProjectClustersService.java

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.core;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.frigga.Names;
+import com.netflix.spinnaker.clouddriver.core.services.Front50Service;
+import com.netflix.spinnaker.clouddriver.model.Cluster;
+import com.netflix.spinnaker.clouddriver.model.ClusterProvider;
+import com.netflix.spinnaker.clouddriver.model.ServerGroup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.inject.Provider;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class ProjectClustersService {
+
+  private final static Logger log = LoggerFactory.getLogger(ProjectClustersService.class);
+
+  private final Front50Service front50Service;
+  private final ObjectMapper objectMapper;
+  private final Provider<List<ClusterProvider>> clusterProviders;
+
+  public ProjectClustersService(Front50Service front50Service,
+                                ObjectMapper objectMapper,
+                                Provider<List<ClusterProvider>> clusterProviders) {
+    this.front50Service = front50Service;
+    this.objectMapper = objectMapper.copy().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    this.clusterProviders = clusterProviders;
+  }
+
+  public Map<String, List<ClusterModel>> getProjectClusters(List<String> projectNames) {
+    List<Map> projects = front50Service.searchForProjects(Collections.emptyMap(), 1000);
+
+    Map<String, List<ProjectClustersService.ClusterModel>> projectClusters = new ConcurrentHashMap<>();
```

Comment:
`ConcurrentHashMap`? Why?

Reply:
Mmm, that's a leftover - will make it a normal HashMap, thanks.